packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -696,7 +931,13 @@ const VirtualListBaseFactory = (type) => {
 
 				if (++j === dimensionToExtent) {
 					secondaryPosition = 0;
-					primaryPosition += primary.gridSize;
+					if (childPositionInfo[i]) {
+						primaryPosition = childPositionInfo[i].position;
+					} else if (childPositionInfo[i - dimensionToExtent]) {
```

Comment:
Is it efficient to pass not an item index but an extent as an index of `childPositionInfo`? Then we could use more cached position.

Reply:
All items in the same extent have the same primary position, so we don't need to cache info for each item but for each extent. So, cached info for the first item of an extent.